src/FullCalendar.tsx

```diff
@@ -1,32 +1,22 @@
 import * as React from 'react'
 import { Calendar, OptionsInput } from '@fullcalendar/core'
 
-export default class FullCalendar extends React.Component<OptionsInput, any> {
+const { forwardRef, useEffect, useImperativeHandle, useRef, useState } = React
 
-  private elRef: any = React.createRef()
-  private calendar: Calendar
+export default forwardRef((props: OptionsInput, componentRef) => {
+  const elRef = useRef()
+  const [calendar, setCalendar] = useState()
 
-  render() {
-    return (
-      <div ref={this.elRef}></div>
-    )
-  }
+  useEffect(() => {
+    const c = new Calendar(elRef.current, props)
+    setCalendar(c)
+    c.render()
+    return () => c.destroy()
+  }, [])
```

Comment:
can you add a comment that explains that the second arg is doing? same with all other similar calls where this second arg functions as a cachebuster essentially

Reply:
I sure can explain the purpose of the second arg but it's basic hook stuff; nothing really special. Maybe adding a comment above each hook explaining what will happen is better instead of clarifying the second arguments?

  ```
// Calendar initialization
  useEffect(() => {
    ...
  }, [])

  // Calendar options update when props change
  useEffect(() => calendar && calendar.resetOptions(props), [calendar, props])

  // Allow getApi function to be exposed to parent component when using ref
  useImperativeHandle(componentRef, () => ({
    getApi: (): Calendar => calendar
  }), [calendar])
```

I don't know what Calendar.destroy internally do. Thus, I ask you if it is really necessary to call it before the the wrapper component is destroyed. Otherwise this kind of comment would do: 
```
// Cleanup callback automatically called before unmounting
return () => c.destroy()
```

Reply:
sounds good